sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/ServiceBusMessageConverter.cs

```diff
@@ -0,0 +1,120 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Azure.Messaging.ServiceBus;
+
+namespace Microsoft.Azure.WebJobs.ServiceBus
+{
+    internal class ServiceBusMessageConverter : JsonConverter<ServiceBusMessage>
```

Comment:
It may be helpful to have some unit tests for different scenarios rather than relying on the end-to-end tests.  That may help to improve the dev loop and better isolate a specific regression since the E2E test has so much going on and needs to be run Live.